clang/lib/Sema/SemaLookup.cpp

```diff
@@ -893,8 +895,15 @@ static void InsertBuiltinDeclarationsFromTable(
 
     // Create function overload for each type combination.
     std::vector<QualType> FunctionList;
+    ExceptionSpecificationType EST =
+        (S.getLangOpts().DeclareSPIRVBuiltins &&
+         (S.getLangOpts().SYCLIsDevice || S.getLangOpts().OpenMPIsTargetDevice))
+            ? EST_BasicNoexcept
+            : EST_None;
```

Comment:
These builtins won't raise an exception by specs, so I think it is safe to always set to noexcept

Reply:
thanks for the suggestion, I've unconditionally set EST_BasicNoexcept, for both OpenCL and SPIR-V builtins. Please check again.